packages/fxa-auth-server/lib/luaScripts/schema.md

```diff
@@ -0,0 +1,126 @@
+# Schema for data stored in Redis
+
+We currently store three different types of data in redis:
+
+- high-churn sessionToken metadata
+- short-lived OAuth access tokens
+- high-churn refreshToken metadata
+
+Each type of data is stored under a different (and configurable) key prefix
+to avoid collisions.
+
+## Session Token Metadata
+
+Most sessionToken data is stored in MySQL and is updated rarely, but a small
+number of fields may change much more often. Writes to these fields are sent
+to redis because it's better able to cope with the write load, and there are
+no significant consequences to losing the data.
+
+The data is stored as a single record per user, using their `uid` as the key.
+The value is a JSON object whose keys are the ids of session tokens belonging
+to that user, and whose corresponding values are the metadata for that token.
+The metadata is a JSON **_list_** where each item corresponds to a particular
+field of the session token.
+
+For example, if a user has a single session token with the following metadata:
+
+- `id`: "000111222333444555666777888999"
+- `lastAccessTime`: 1583392038878
+- `location`:
+  - city: undefined,
+  - state: undefined
+  - stateCode: undefined,
+  - country: "Australia"
+  - countryCode: "AU"
+- uaBrowser: undefined,
+- uaBrowserVersion: undefined,
+- uaOS: "Windows",
+- uaOSVersion: 10,
+- uaDeviceType: undefined,
+- uaFormFactor: undefined
+
+Then this would be stored in redis as a string value as follows:
+
+- Key: `000111222333444555666777888999`
+- Value: `[1583392038878,[,,,"Australia","AU"],,,"Windows",10,,]`
```

Comment:
Every time I see this, I regret this serialization scheme O_O

Reply:
Indeed. We could phase it out easily enough. At least it's encapsulated in redis now.